linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/IstioIdentifier.scala

```diff
@@ -23,14 +24,36 @@ class IstioIdentifier(pfx: Path, baseDtab: () => Dtab, routeCache: RouteCache, c
     }
   }
 
+  def compareStringMatch(headerValue: String, stringMatch: StringMatch): Boolean = {
+    stringMatch.`matchType` match {
+      case Some(OneofMatchType.Exact(value)) => headerValue == value
+      case Some(OneofMatchType.Prefix(pfx)) => headerValue.startsWith(pfx)
+      case Some(OneofMatchType.Regex(r)) => headerValue.matches(r)
+      case None => throw new IllegalArgumentException("stringMatch missing matchType")
+    }
+  }
+
+  def compareMatchConditions(req: Request, matchCondition: MatchCondition): Boolean = {
+    val matchesHeaders = matchCondition.`httpHeaders`.map {
+      case (headerName, stringMatch) =>
+        println(headerName, stringMatch)
+        // return false if the headerName does not appear in the request's headerMap
+        req.headerMap.get(headerName).map(compareStringMatch(_, stringMatch)).getOrElse(false)
```

Comment:
could also do
```
req.headerMap.get(headerName) match {
  case Some(headerValue) => headerMatches(headerValue, stringMatch)
  case None => false
}
```

(.get.map.getOrElse is also fine, just throwing this out there.  go with whatever seems clearer to you)